packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -35,58 +35,49 @@ export const accordionHeaderShorthandProps: Array<keyof AccordionHeaderSlots> =
  */
 export const useAccordionHeader = (props: AccordionHeaderProps, ref: React.Ref<HTMLElement>): AccordionHeaderState => {
   const { onHeaderClick: onAccordionHeaderClick, disabled, open } = useAccordionItemContext();
-  const button = useContextSelector(AccordionContext, ctx => ctx.button);
-  const expandIcon = useContextSelector(AccordionContext, ctx => ctx.expandIcon);
-  const inline = useContextSelector(AccordionContext, ctx => ctx.inline);
-  const icon = useContextSelector(AccordionContext, ctx => ctx.icon);
-  const expandIconPosition = useContextSelector(AccordionContext, ctx => ctx.expandIconPosition);
-  const size = useContextSelector(AccordionContext, ctx => ctx.size);
   const id = useId('accordion-header-', props.id);
   const panel = useDescendants(accordionItemDescendantContext)[1] as AccordionItemDescendant | undefined;
   const innerRef = React.useRef<HTMLElement>(null);
-  const initialState = {
+
+  useAccordionItemDescendant(
+    {
+      element: innerRef.current,
+      id,
+    },
+    0,
+  );
+
+  const originalHandleClick = isObjectShorthandProps(props.button) ? props.button.onClick : undefined;
+
+  return {
     ...props,
+    disabled,
+    open,
     ref: useMergedRefs(ref, innerRef),
-    size: props.size ?? size ?? 'medium',
-    inline: props.inline ?? inline ?? false,
-    role: props.role ?? 'heading',
-    expandIconPosition: props.expandIconPosition ?? expandIconPosition ?? 'start',
+    size: props.size || 'medium',
+    inline: props.inline || false,
+    role: props.role || 'heading',
+    expandIconPosition: props.expandIconPosition || 'start',
     components: {
       root: 'div',
       button: 'button',
       expandIcon: AccordionHeaderExpandIcon,
     },
-    icon: resolveShorthand(props.icon ?? icon),
-    expandIcon: resolveShorthand(props.expandIcon ?? expandIcon, {
+    icon: resolveShorthand(props.icon),
+    expandIcon: resolveShorthand(props.expandIcon, {
       'aria-hidden': true,
     }),
-    button: useARIAButton(props.button ?? button, {
+    button: useARIAButton(props.button, {
       id,
-      onClick: onAccordionHeaderClick,
+      onClick: useEventCallback((ev: React.MouseEvent<HTMLElement>) => {
+        originalHandleClick?.(ev);
+        if (!ev.isDefaultPrevented()) {
```

Comment:
Curious why is this check needed ? accordion behaviour is not a part of the default click behaviour

Reply:
The way I imagined is that opening and closing the `AccordionItem` is the default behavior of the button that belongs to `AccordionItem`. Otherwise there would be no way to cancel opening and closing.

Thinking about that this my be redundant for the case of controlled value. 🤔

Reply:
That's a fair point, but I as you said, we do support controlling so it seems redundant to me too